async-nats/src/lib.rs

```diff
@@ -1397,3 +1415,15 @@ impl<T: ToServerAddrs + ?Sized> ToServerAddrs for &T {
         (**self).to_server_addrs()
     }
 }
+
+#[derive(Clone)]
+pub(crate) enum Authorization {
+    /// No authentication.
+    None,
+
+    /// Authenticate using a token.
+    Token(String),
+
+    /// Authenticate using a username and password.
+    UsernamePassword(String, String),
```

Comment:
Awkward name imo

```suggestion
    Credentials(String, String),
```

Or `Basic` (ala http basic authorization)

Reply:
I have to again disagree here, again 😄 

`Credentials` would be easily confused with method that we have to add in the future that implements loading NATS `.creds` files.

Reply:
Option C then 😁